rskj-core/src/test/java/co/rsk/rpc/netty/Web3WebsocketServerTest.java

```diff
@@ -0,0 +1,109 @@
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.squareup.okhttp.*;
+import com.squareup.okhttp.ws.WebSocket;
+import com.squareup.okhttp.ws.WebSocketCall;
+import com.squareup.okhttp.ws.WebSocketListener;
+import okio.Buffer;
+import org.ethereum.rpc.Web3;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.*;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class Web3WebsocketServerTest {
+
+    private static JsonNodeFactory JSON_NODE_FACTORY = JsonNodeFactory.instance;
+    private static ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    @Test
+    public void smokeTest() throws Exception {
+        Web3 web3Mock = mock(Web3.class);
+        String mockResult = "output";
+        when(web3Mock.web3_sha3(anyString())).thenReturn(mockResult);
+
+        int randomPort = 9998;//new ServerSocket(0).getLocalPort();
+
+        List<ModuleDescription> filteredModules = Collections.singletonList(new ModuleDescription("web3", "1.0", true, Collections.emptyList(), Collections.emptyList()));
+        JsonRpcWeb3ServerHandler serverHandler = new JsonRpcWeb3ServerHandler(web3Mock, filteredModules);
+
+        Web3WebsocketServer websocketServer = new Web3WebsocketServer(InetAddress.getLoopbackAddress(), randomPort, serverHandler);
+        websocketServer.start();
+
+        OkHttpClient wsClient = new OkHttpClient();
+        Request wsRequest = new Request.Builder().url("ws://localhost:"+randomPort+"/websocket").build();
+        WebSocketCall wsCall = WebSocketCall.create(wsClient, wsRequest);
+
+        CountDownLatch wsAsyncResultLatch = new CountDownLatch(1);
+        wsCall.enqueue(new WebSocketListener() {
+            @Override
+            public void onOpen(WebSocket webSocket, Response response) {
+                RequestBody body = RequestBody.create(WebSocket.TEXT, getJsonRpcDummyMessage());
+                try {
+                    webSocket.sendMessage(body);
+                } catch (IOException e) {
+                    fail(e.getMessage());
+                }
+            }
+
+            @Override
+            public void onFailure(IOException e, Response response) {
+                fail(e.getMessage());
+            }
+
+            @Override
+            public void onMessage(ResponseBody message) throws IOException {
+                JsonNode jsonRpcResponse = OBJECT_MAPPER.readTree(message.bytes());
+                assertThat(jsonRpcResponse.at("/result").asText(), is(mockResult));
```

Comment:
Maybe add another check to see that this is always called. I'm worried `onMessage` might never execute and the test would still pass.

Reply:
This is the only method counting down the _latch_, if that doesn't happen in 10 seconds, the test fails